shared/src/commonMain/kotlin/io.newm.shared/config/NewmSharedBuildConfig.kt

```diff
@@ -0,0 +1,57 @@
+package io.newm.shared.config
+
+import com.liftric.kvault.KVault
+import io.newm.shared.generated.SharedBuildConfig
+import org.koin.core.component.KoinComponent
+import org.koin.core.component.inject
+
+// Enum to represent the mode in which the app is running
+enum class Mode {
+    PRODUCTION,
+    DEVELOPMENT
+}
+
+// BuildConfiguration class to manage app configurations based on the mode
+object NewmSharedBuildConfig: KoinComponent {
+
+    private const val APP_MODE = "app_mode"
+
+    private val storage: KVault by inject()
+    var mode: Mode
+        get() {
+            val modeString = storage.string(APP_MODE) ?: Mode.PRODUCTION.name
+            return Mode.valueOf(modeString)
+        }
+        set(value) {
+            storage.set(APP_MODE, value.name)
+        }
+
+    init {
+        // If there's no mode saved in KVault, default to Production.
+        if (storage.string(APP_MODE) == null) {
+            mode = Mode.PRODUCTION
+        }
+    }
+
+    val baseUrl: String
+        get() = when (mode) {
+            Mode.DEVELOPMENT -> SharedBuildConfig.DEVELOPMENT_URL
+            Mode.PRODUCTION -> SharedBuildConfig.PRODUCTION_URL
+        }
+
+    val sentryAuthToken: String
+        get() = SharedBuildConfig.SENTRY_AUTH_TOKEN
+
+    val googleAuthClientId: String
+        get() = SharedBuildConfig.GOOGLE_AUTH_CLIENT_ID
+
+    val walletConnectProjectId: String
+        get() = SharedBuildConfig.WALLET_CONNECT_PROJECT_ID
+
+    val recaptchaSiteKey: String
+        get() = SharedBuildConfig.RECAPTCHA_SITE_KEY
+
+    val isDevMode: Boolean
+        get() = mode == Mode.DEVELOPMENT
+}
```

Comment:
Why are these in kmm if they're not used in kmm?  (besides walletConnectProjectId)

Reply:
iOS needs some of these at compile time, but this object is only available at runtime.

Reply:
How did you verify they're available at compile time in iOS as an environment variable?  That's how they're used.

Reply:
I ran the iOS app on Android studio and it worked fine. You don't have to use the same variables if you don't like, you can continue using the secrets file for now.